url64d634b8f0_agdturner_generic_tgz-pJ8-uk_ac_leeds_ccg_andyt_generic_visualisation_charts_Generic_BarChartJ8/src/uk/ac/leeds/ccg/andyt/generic/io/Generic_StaticIO.java

```diff
@@ -0,0 +1,2156 @@
+//////                                a_DirectoryToCopy_File_Files[i],
+//////                                a_DirectoryToCopyInto_2_File);
+//////                    } else {
+//////                        if (!a_DirectoryToCopyInto_2_File.mkdir()) {
+//////                            a_DirectoryToCopyInto_2_File.mkdirs();
+//////                        }
+//////                        copy(
+//////                                a_DirectoryToCopy_File_Files[i],
+//////                                a_DirectoryToCopyInto_2_File);
+//////                    }
+////                    File a_DirectoryToCopyInto_2_File = new File(
+////                            a_DirectoryToCopyInto_1_File,
+////                            a_DirectoryToCopy_File_Files[i].getName());
+//                    if (a_DirectoryToCopy_File_Files[i].isFile()) {
+//                        copyFile(
+//                                a_DirectoryToCopy_File_Files[i],
+//                                a_DirectoryToCopyInto_1_File);
+//                    } else {
+//                        if (!a_DirectoryToCopyInto_1_File.mkdir()) {
+//                            a_DirectoryToCopyInto_1_File.mkdirs();
+//                        }
+//                        copy(
+//                                a_DirectoryToCopy_File_Files[i],
+//                                a_DirectoryToCopyInto_1_File);
+//                    }
+//                }
+            }
+        } catch (IOException e) {
+            System.err.print(e.getMessage());
+            e.printStackTrace(System.err);
+            System.exit(Generic_ErrorAndExceptionHandler.IOException);
+        }
+    }
+
+    /**
+     * @return true iff file exists and can be read
+     * @param file
+     */
+    public static boolean fileExistsAndCanBeRead(File file) {
+        if (file.exists()) {
+            if (file.canRead()) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Attempts to delete file from the file system. This does no block other
+     * reads or writes to the files system while it is executing and it will
+     * only succeed in deleting empty directories.
+     *
+     * @param file
+     * @return long[] result where: result[0] is the number of directories
+     * deleted; result[1] is the number of files deleted.
+     */
+    public static long[] delete(File file) {
+        long[] result = new long[2];
+        result[0] = 0L;
+        result[1] = 0L;
+        if (file.isDirectory()) {
+            // Delete all the files it contains
+            File[] files = file.listFiles();
+            for (File file1 : files) {
+                long[] deleted = delete(file1);
+                result[0] += deleted[0];
+                result[1] += deleted[1];
+            }
+            // Delete the directory itself
+            file.delete();
+            result[0]++;
+        } else {
+            file.delete();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The result of deleting a file on this line should be checked and failure should be logged. java.io.File.delete() will simply return false if it fails to delete the file, for example, when the file does not exist. It will not throw unless there is a security exception.